frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -0,0 +1,404 @@
+              }}
+            >
+              Add another child
+            </Button>
+            <Spacer />
+            <Text
+              textStyle="body-2"
+              color="secondary.400"
+            >{simplur`${numChild} child[|ren] added`}</Text>
+          </HStack>
+        ) : null}
+      </VStack>
+    </FieldContainer>
+  )
+}
+
+interface ChildrenBodyProps {
+  currChildBodyIdx: number
+  schema: ChildrenCompoundFieldSchema
+  fields: FieldArrayWithId<
+    ChildrenCompoundFieldInputs,
+    `${string}.child`,
+    'id'
+  >[]
+  field: FieldArrayWithId<ChildrenCompoundFieldInputs, `${string}.child`, 'id'>
+  colorTheme: FormColorTheme
+  remove: UseFieldArrayRemove
+  myInfoChildrenBirthRecords?: MyInfoChildData
+  isSubmitting: boolean
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  formContext: UseFormReturn<ChildrenCompoundFieldInputs, any>
+}
+
+const ChildrenBody = ({
+  currChildBodyIdx,
+  schema,
+  fields,
+  field,
+  colorTheme,
+  remove,
+  myInfoChildrenBirthRecords,
+  isSubmitting,
+  formContext,
+}: ChildrenBodyProps): JSX.Element => {
+  const { register, getValues, setValue, watch } = formContext
+
+  const childNamePath = useMemo(
+    () => `${schema._id}.child.${currChildBodyIdx}.0`,
+    [schema._id, currChildBodyIdx],
+  )
+  const { onChange: selectOnChange, ...selectRest } = register(childNamePath)
+
+  const childName = watch(childNamePath) as unknown as string
+
+  const allChilds = useMemo<string[]>(() => {
+    if (myInfoChildrenBirthRecords === undefined) {
+      return []
+    }
+    return myInfoChildrenBirthRecords[MyInfoAttribute.ChildName] ?? []
+  }, [myInfoChildrenBirthRecords])
+
+  // useCallback to re-compute names because for some reason watch doesn't
+  // work on this nested field in react-hook-form.
+  const allSelectedNames = useCallback((): string[] => {
+    // Really important to note that sometimes react-hook-form stores our "array"
+    // as a object with key=index and values=array entry.
+    return Object.values(getValues(`${schema._id}.child`)).map((arr) => arr[0])
+  }, [getValues, schema._id])
+
+  // useCallback to re-compute names, again because of buggy allSelectedNames
+  const namesNotSelected = useCallback((): string[] => {
+    if (myInfoChildrenBirthRecords === undefined) {
+      return []
+    }
+    const temp = allSelectedNames()
+    // We want all child names that haven't already been selected.
+    // O(n^2) but n is small so it should be okay.
+    return allChilds.filter((name) => !temp.includes(name))
+  }, [myInfoChildrenBirthRecords, allChilds, allSelectedNames])
+
+  const getChildAttr = useCallback(
+    (attr: MyInfoChildAttributes): string => {
+      if (myInfoChildrenBirthRecords === undefined) {
+        return ''
+      }
+      // First find the child
+      const indexOfChild =
+        myInfoChildrenBirthRecords[MyInfoChildAttributes.ChildName]?.indexOf(
+          childName,
+        )
```

Comment:
So in small set sizes, this is probably very fast, but still, considering we know all the subfields are iterated on later on, we could have collated all the fields from the current child at bootstrap with a useMemo(), and then `getChildAttr()` could do a direct lookup on it rather than have to find the child index again for every subfield.